cps/kobo.py

```diff
@@ -0,0 +1,508 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#  This file is part of the Calibre-Web (https://github.com/janeczku/calibre-web)
+#    Copyright (C) 2018-2019 OzzieIsaacs
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+import uuid
+from base64 import b64decode, b64encode
+from datetime import datetime
+from time import gmtime, strftime
+try:
+    from urllib import unquote
+except ImportError:
+    from urllib.parse import unquote
+
+from jsonschema import validate, exceptions
+from flask import (
+    Blueprint,
+    request,
+    make_response,
+    jsonify,
+    json,
+    current_app,
+    url_for,
+    redirect,
+)
+from flask_login import login_required
+from werkzeug.datastructures import Headers
+from sqlalchemy import func
+import requests
+
+from . import config, logger, kobo_auth, db, helper
+from .web import download_required
+
+KOBO_FORMATS = {"KEPUB": ["KEPUB"], "EPUB": ["EPUB", "EPUB3"]}
+KOBO_STOREAPI_URL = "https://storeapi.kobo.com"
+
+kobo = Blueprint("kobo", __name__, url_prefix="/kobo/<auth_token>")
+kobo_auth.disable_failed_auth_redirect_for_blueprint(kobo)
+kobo_auth.register_url_value_preprocessor(kobo)
+
+log = logger.create()
+
+
+def b64encode_json(json_data):
+    if sys.version_info < (3, 0):
+        return b64encode(json.dumps(json_data))
+    else:
+        return b64encode(json.dumps(json_data).encode())
+
+
+# Python3 has a timestamp() method we could be calling, however it's not avaiable in python2.
+def to_epoch_timestamp(datetime_object):
+    return (datetime_object - datetime(1970, 1, 1)).total_seconds()
+
+
+def get_store_url_for_current_request():
+    # Programmatically modify the current url to point to the official Kobo store
+    base, sep, request_path_with_auth_token = request.full_path.rpartition("/kobo/")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'sep'](https://app.codacy.com/manual/Ozzie-IsaacsOrganization/calibre-web/pullRequest?prid=4846308)

Reply:
I personally prefer leaving these named since they document the url format and rpartition return values.